### PR TITLE
Usage agreement form image changes

### PR DESCRIPTION
**Pull request by aparajit-pratap**
@Benglin this PR has incremental changes for the usage agreement form image as well as refactoring of `Bitmap` to `BitmapSource` conversion API.

### CODE REVIEW

File: src/DynamoCoreWpf/Utilities/ResourceUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System.Windows.Interop;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+
+namespace Dynamo.Utilities
+{
+    public static class ResourceUtilities
+    {
+
+        public static ImageSource SetImageSource(System.Drawing.Bitmap bmp)
```
Comment (Benglin) [inline src/DynamoCoreWpf/Utilities/ResourceUtilities.cs]: Like I mentioned in another PR, this should really be called `ConvertToImageSource`.

File: src/DynamoCoreWpf/Services/UsageReportingManager.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@ public class UsageReportingManager : NotificationObject
 
         private static DynamoModel dynamoModel;
         private static UsageReportingManager instance;
+        private static IBrandingResourceProvider resourceProvider;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Services/UsageReportingManager.cs]: Hang on, what's this for?

**Comment by Benglin** [comment]
Almost perfect, just need to confirm that very critical thing.

**Comment by aparajit-pratap** [comment]
@Benglin I have made the necessary code changes as well as tested the watch image node that could be impacted by this change. It should be good to go now. Thanks.

**Comment by lukechurch** [comment]
@RodRecker

**Comment by RodRecker** [comment]
I'm ok taking this now with the understanding that the text of the agreement will likely change soon.

**Comment by aparajit-pratap** [comment]
@RodRecker yes the text and the images are simply placeholders for now. They will be replaced with the new ones later. These code changes simply provide the infrastructure to make the swap. Thanks.

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/WatchImage.cs [hunk 3]
```diff
@@ -44,22 +42,18 @@ public void CustomizeView(WatchImageCore nodeModel, NodeView nodeView)
...
-                return;
-
-            // how to convert a bitmap to an imagesource http://blog.laranjee.com/how-to-convert-winforms-bitmap-to-wpf-imagesource/ 
-            // TODO - watch out for memory leaks using system.drawing.bitmaps in managed code, see here http://social.msdn.microsoft.com/Forums/en/csharpgeneral/thread/4e213af5-d546-4cc1-a8f0-462720e5fcde
-            // need to call Dispose manually somewhere, or perhaps use a WPF native structure instead of bitmap?
-
-            var hbitmap = bmp.GetHbitmap();
-            var imageSource = Imaging.CreateBitmapSourceFromHBitmap(hbitmap, IntPtr.Zero, Int32Rect.Empty, BitmapSizeOptions.FromWidthAndHeight(bmp.Width, bmp.Height));
-            image.Source = imageSource;
+            image.Source = ResourceUtilities.ConvertToImageSource(bmp);
```
Comment (Benglin) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/WatchImage.cs]: Hi @aparajit-pratap, let's try to keep that set image done asynchronously, if at all possible (we have seen twice a deadlock condition when main window is closed, the main thread tries to shut down `ISchedulerThread` while the thread is waiting for main thread to process this `Invoke` synchronous call. Here's what I'd recommend:

```csharp
private void NodeModelOnPropertyChanged(object sender, PropertyChangedEventArgs args)
{
    if (args.PropertyName != "IsUpdated")
        return;

    var data = nodeModel.CachedValue;
    var bitmap = data.Data as Bitmap;
    if (bitmap != null)
    {
        nodeView.Dispatcher.BeginInvoke(new Action<Bitmap>(SetImageSource),
        new object[] { nodeModel.CachedValue.Data as Bitmap });
    }
}

private void SetImageSource(System.Drawing.Bitmap bmp)
{
    if (bmp != null)
    {
        using (bmp)
        {
            image.Source = ResourceUtilities.ConvertToImageSource(bmp);
        }
    }
}
```

Please have a go and let's see where that leads us. Thanks!

**Comment by aparajit-pratap** [comment]
@Benglin done!

**Comment by Benglin** [comment]
Thanks for doing this, @aparajit-pratap! LGTM!

**Pull request merged**